it/uk/gov/hmrc/agentmappingfrontend/support/EndpointBehaviours.scala

```diff
@@ -28,80 +28,62 @@ trait EndpointBehaviours extends AuthStubs {
       auditEventShouldNotHaveBeenSent("CheckAgentRefCode")
     }
 
-    "redirect to the start page if the current user is logged in and does not have affinity group Agent" in {
-      givenUserIsAuthenticated(individual)
+    "redirect to the sign-in page if the current user is logged in but does not have affinity group Agent" in {
+      givenUnauthorisedWith("UnsupportedAffinityGroup")
       val request = fakeRequest(endpointMethod,endpointPath)
       val result = await(doRequest(request))
 
       result.header.status shouldBe 303
-      result.header.headers("Location") shouldBe routes.MappingController.notEnrolled().url
-
+      result.header.headers("Location") should include("/gg/sign-in")
       auditEventShouldNotHaveBeenSent("CheckAgentRefCode")
     }
   }
 
-  protected def anEndpointReachableGivenAgentAffinityGroupAndIrSaAgentEnrolment(endpointMethod: String, endpointPath: String,
-                                                                                expectCheckAgentRefCodeAudit: Boolean)
-                                                                               (doRequest: FakeRequest[AnyContentAsEmpty.type] => Result): Unit = {
-    "redirect to the company-auth-frontend sign-in page if the current user is not logged in" in {
-      givenUserIsNotAuthenticated()
-
-      val request = fakeRequest(endpointMethod, endpointPath)
-      val result = await(doRequest(request))
-
-      result.header.status shouldBe 303
-      result.header.headers("Location") should include("/gg/sign-in")
-      auditEventShouldNotHaveBeenSent("CheckAgentRefCode")
-    }
+  protected def anEndpointReachableGivenAgentAffinityGroupAndValidEnrolment(endpointMethod: String, endpointPath: String,
+                                                                            expectCheckAgentRefCodeAudit: Boolean)
+                                                                           (doRequest: Request[AnyContentAsEmpty.type] => Result): Unit = {
+    behave like anAuthenticatedEndpoint(endpointMethod, endpointPath, doRequest)
 
-    "redirect to the start page if the current user is logged in and does not have affinity group Agent" in {
-      givenUserIsAuthenticated(individual)
+    "render the not-enrolled page if the current user is logged with affinity group Agent but is not validly enrolled" in {
+      givenUserIsAuthenticated(aStrangeAgent)
```

Comment:
aStrangeAgent? name of this doesn't really explain much, what is strange.
you can loose the a's, and just use strangeAgent